### PR TITLE
Rounds backend functionality

### PR DESCRIPTION
**Pull request by boczkas**
Backend functionality for rounds introduced.
Round is not blocked yet - front end work ongoing.
You can review front end code from clean code point of view, without functionality check.

There are some places in which constant room name is present - it is part of work connected with future rooms functionality.

### CODE REVIEW

File: BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/player/PlayingPlayer.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        this.player = player;
+    }
+
+    @Override
+    public String toString() {
+        return "PlayingPlayer{" +
+                "player=" + player +
+                '}';
+    }
+}
```
Comment (JaroslawSlaby) [inline BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/player/PlayingPlayer.java]: equals() and hashCode()

File: BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerConfigurationEventListener.java [hunk 2]
```diff
@@ -25,5 +30,9 @@ public void onApplicationEvent(PlayerConfigurationEvent event) {
         boardHandler.addPlayer(player);
         String roomName = "roomName";
         boardHandler.addPlayerToRoom(new Room(roomName), player);
+
+        if(boardHandler.areOpponentsPresent(roomName)){
```
Comment (totine) [inline BoardHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/PlayerConfigurationEventListener.java]: space

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/utils/SendMessage.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package pl.jakubowskiprzemyslaw.REST.utils;
+
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SendMessage {
```
Comment (totine) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/utils/SendMessage.java]: MessageSender

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/utils/SendMessage.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    private final SimpMessagingTemplate template;
+
+    public SendMessage(SimpMessagingTemplate template) {
+        this.template = template;
+    }
+
+    public void sendMessage(String playerName, String message) {
```
Comment (totine) [inline REST/src/main/java/pl/jakubowskiprzemyslaw/REST/utils/SendMessage.java]: mention about clock

File: REST/src/main/resources/static/js/board.js [hunk 5]
```diff
@@ -1,9 +1,18 @@
 var myBoard = document.getElementById("myBoard");
 var opponentBoard = document.getElementById("opponentBoard");
+var timerIntervalID;
+var myRound = true;
+var stompClient = null;
 
 window.onload = function load() {
     setDivsForBoard(myBoard, false, true);
```
Comment (totine) [inline REST/src/main/resources/static/js/board.js]: weird trues and falses

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/configuration/WebSocketConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        registry.setUserDestinationPrefix("/user");
+    }
+
+    @Override
+    public void registerStompEndpoints(StompEndpointRegistry registry) {
+        registry.addEndpoint("/timer-websocket").withSockJS();
+    }
+
+    @Override
+    public boolean configureMessageConverters(List<MessageConverter> messageConverters) {
```
[the remark this answers is not in the document]
Comment (boczkas) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/configuration/WebSocketConfiguration.java]: wasn't needed

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/utils/SendMessage.java (hunk 3 above)
Comment (boczkas) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/utils/SendMessage.java]: it's done already, will be shown in next PR

File: REST/src/main/java/pl/jakubowskiprzemyslaw/REST/utils/SendMessage.java (hunk 4 above)
Comment (boczkas) [reply REST/src/main/java/pl/jakubowskiprzemyslaw/REST/utils/SendMessage.java]: it's done already, will be shown in next PR

File: REST/src/main/resources/static/js/board.js (hunk 5 above)
Comment (boczkas) [reply REST/src/main/resources/static/js/board.js]: it's done already, will be shown in next PR

**Pull request merged**